### PR TITLE
:fix switch network improvements & wagmi to recognize wc session

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Description

Resolve cases where wagmi doesn't recognize wc session and unable to switch from unsupported network"

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes [APKT-xxx](https://linear.app/reown/issue/DEC-256/connecting-wallet-thats-not-already-on-op-sepolia-results-in-no)
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [ ] My changes generate no new warnings
- [ ] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/wagmi/src/connectors/UniversalConnector.ts [hunk 1]
```diff
@@ -276,14 +276,14 @@ export function walletConnect(
...
-        }
-
         await provider.request({
           method: 'wallet_switchEthereumChain',
           params: [{ chainId: numberToHex(chainId) }]
         })
+
+        if (chainToSwitch?.caipNetworkId) {
+          provider.setDefaultChain(chainToSwitch?.caipNetworkId as string)
+        }
```
Comment (tomiir) [inline packages/adapters/wagmi/src/connectors/UniversalConnector.ts:286]: Why move to after?  How does it change things?

File: packages/appkit/src/client.ts [hunk 2]
```diff
@@ -1282,10 +1283,12 @@ export class AppKit {
...
           } catch (error) {
-            adapter?.switchNetwork({
-              provider: this.universalProvider,
-              caipNetwork: ChainController.state.activeCaipNetwork as CaipNetwork
-            })
+            if (adapter?.reconnect) {
+              adapter?.reconnect({
+                id: 'walletConnect',
+                type: 'WALLET_CONNECT'
+              })
```
Comment (tomiir) [inline packages/appkit/src/client.ts:1290]: can we add a comment on why this is required? not sure why this or a switch network would be required if a connection fails? 🤔

File: packages/adapters/wagmi/src/connectors/UniversalConnector.ts (hunk 1 above)
Comment (svenvoskamp) [reply packages/adapters/wagmi/src/connectors/UniversalConnector.ts:286]: For some wallets (e.g MM) will emit a `chainChanged` event when this is called and the chainId is not the active chainId. So thats why we first need to switch to the correct chain.

File: packages/appkit/src/client.ts (hunk 2 above)
Comment (svenvoskamp) [reply packages/appkit/src/client.ts:1294]: Added a comment:

`Handle edge case where wagmi detects existing connection but lacks to complete UniversalProvider instance. Connection attempt fails due to already connected state - reconnect to restore provider state.`

**Review by tomiir — APPROVED** [review]

**Pull request merged**